src/commands/mod.rs

```diff
@@ -289,6 +289,70 @@ impl DaemonControl {
         GetAddressResult::new(address)
     }
 
+    /// list addresses
+    pub fn list_addresses(
+        &self,
+        start_index: Option<u32>,
+        count: Option<u32>,
+    ) -> ListAddressesResult {
+        let mut db_conn = self.db.connection();
+        let receive_index: u32 = db_conn.receive_index().into();
+        let change_index: u32 = db_conn.change_index().into();
+
+        let start_index = start_index.unwrap_or(0);
+
+        let index;
+        match count {
+            None => {
+                let max_index = receive_index.max(change_index);
+                index = if start_index >= max_index {
+                    start_index
+                } else {
+                    if max_index > 0 {
+                        max_index - 1
+                    } else {
+                        0
+                    }
+                };
+            }
+            Some(c) => {
+                index = if (start_index + c) > 0 {
+                    start_index + c - 1
+                } else {
+                    0
+                }
+            }
+        }
```

Comment:
not sure if a match is the better way to handle this ?

Reply:
Perhaps as you are only interested in `Some(c)` or `None` you could use `if let Some(c) = count {} else {}`.

Nit: for the subtraction logic, you could try [saturating_sub](https://doc.rust-lang.org/std/primitive.u32.html#method.saturating_sub) instead of the `if/else`.

Reply:
thx a lot!